More optimizations

trask opened the pull request.

littleaj commented on core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java:
```diff
@@ -292,40 +307,30 @@ private void writeToJson(String name, String value, int len) throws IOException
-        String jsonStringToAppend = stringWriter.toString();
-        if (Strings.isNullOrEmpty(jsonStringToAppend) || JSON_EMPTY_OBJECT.equals(jsonStringToAppend)) {
-            return "";
+    private <T> void writeObject(JsonSerializable value) throws IOException {
```
Looks like we don't need the type variable anymore

trask replied:
👍

littleaj commented on agent-codeless/src/main/java/com/microsoft/applicationinsights/agentc/internal/model/IncomingSpanImpl.java:
```diff
@@ -227,7 +217,7 @@ private void finishBuildingWebTelemetry(long endTimeMillis) {
         }
 
-        processCookies(detail);
+        // processCookies(detail);
```
should this be removed?

trask replied:
👍

littleaj commented on core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java:
```diff
@@ -31,6 +31,7 @@
 
 import java.io.IOException;
 import java.io.StringWriter;
+import java.lang.ref.WeakReference;
```
Unused

trask replied:
👍

littleaj commented on core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/Envelope.java:
```diff
@@ -45,7 +45,9 @@
-    private String name;
+    private String namePart1;
+    private String namePart2;
+    private String namePart3;
```
It took a minute to find what part 1, 2 and 3 are. Let's name them something like prefix, normalizedIkey, telemetryType or document in javadoc.

trask replied:
👍

littleaj commented on core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java:
```diff
@@ -224,12 +230,7 @@ public static String normalizeInstrumentationKey(String instrumentationKey){
-                normalizedInstrumentationKey,
-                envelopType
-                );
+        return TELEMETRY_NAME_PREFIX + normalizedInstrumentationKey + envelopType;
```
👍

trask commented:
Sorry, I meant to submit this as a draft PR. Some of this work is on top of #1111. I addressed your comments above in #1111, and will come back to this PR once that is merged. Thanks!

trask commented:
Closing, replaced by #1122 and #1123
